renku/cli/_providers/zenodo.py

```diff
@@ -224,6 +229,194 @@ def as_dataset(self):
+        if 'ZENODO_USE_SANDBOX' in os.environ:
+            return self.ZENODO_SANDBOX_URL
+
+        return self.ZENODO_BASE_URL
+
+    def publish_url(self, deposition_id):
+        """Returns publish URL."""
+        return urllib.parse.urlparse((
+            self.zenodo_url + self.API_PATH + self.DEPOSIT_PATH +
+            self.PUBLISH_ACTION_PATH
+        ).format(deposition_id))
+
+    def attach_metadata_url(self, deposition_id):
+        """Return URL for attaching metadata."""
+        return urllib.parse.urlparse((
+            self.zenodo_url + self.API_PATH + self.DEPOSIT_PATH +
+            self.METADATA_URL
+        ).format(deposition_id))
+
+    def upload_file_url(self, deposition_id):
+        """Return URL for uploading file."""
+        return urllib.parse.urlparse((
+            self.zenodo_url + self.API_PATH + self.DEPOSIT_PATH +
+            self.FILES_URL
+        ).format(deposition_id))
+
+    def new_deposit_url(self):
+        """Return URL for creating new deposit."""
+        return urllib.parse.urlparse((
+            self.zenodo_url + self.API_PATH + self.DEPOSIT_PATH +
+            self.NEW_DEPOSIT_URL
+        ))
+
+    def published_at(self, deposition_id):
+        """Return published at URL."""
+        return urllib.parse.urlparse(
+            (self.zenodo_url + self.PUBLISH_PATH + str(deposition_id))
+        )
+
+    def deposit_at(self, deposition_id):
+        """Return deposit at URL."""
+        return urllib.parse.urlparse(
+            (self.zenodo_url + self.DEPOSIT_PATH + str(deposition_id))
+        )
+
+    @staticmethod
+    def _check_or_raise(response):
+        """Check for expected response status code."""
+        if response.status_code not in [200, 201, 202]:
+            if response.status_code == 401:
+                raise HTTPError('Access unauthorized - update access token.')
+
+            if response.status_code == 400:
+                err_response = response.json()
+                errors = [
+                    '"{0}" failed with "{1}"'.format(
+                        err['field'], err['message']
+                    ) for err in err_response['errors']
+                ]
+
+                raise HTTPError('\n' + '\n'.join(errors))
+
+            else:
+                raise HTTPError(response.content)
+
+    @property
+    def default_params(self):
+        """Create request default params."""
+        return {'access_token': self.secret}
+
+    def dataset_to_request(self):
+        """Prepare dataset metadata for request."""
+        jsonld = self.dataset.asjsonld()
+        jsonld['upload_type'] = 'dataset'
+        return jsonld
+
+    def export(self, to_publish):
+        """Execute entire export process."""
+        # Step 1. Create new deposition
+        response = self.new_deposition()
+        ZenodoExporter._check_or_raise(response)
+        response = response.json()
+
+        deposition_id = response['id']
```

Comment:
It might be cleaner to have a separate `ZenodoDeposition` class instantiated via `new_deposition` and the `*_url` methods as properties.